ghost/core/core/server/services/email-analytics/EmailAnalyticsServiceWrapper.js

```diff
@@ -111,30 +111,33 @@ class EmailAnalyticsServiceWrapper {
         }
         this.fetching = true;
 
-        // NOTE: Data shows we can process ~7500 events per minute on Pro; this can vary locally
+        // NOTE: Data shows we can process ~2500 events per minute on Pro for a large-ish db (150k members).
+        //       This can vary locally, but we should be conservative with the number of events we fetch.
         try {
             // Prioritize opens since they are the most important (only data directly displayed to users)
-            await this.fetchLatestOpenedEvents({maxEvents: Infinity});
+            const c1 = await this.fetchLatestOpenedEvents({maxEvents: 10000});
+            if (c1 >= 10000) {
+                this._restartFetch('high opened event count');
+                return;
+            }
 
             // Set limits on how much we fetch without checkings for opened events. During surge events (following newsletter send)
             //  we want to make sure we don't spend too much time collecting delivery data.
-            const c1 = await this.fetchLatestNonOpenedEvents({maxEvents: 20000});
-            if (c1 > 15000) { 
-                this.fetching = false;
-                logging.info('[EmailAnalytics] Restarting fetch due to high event count');
-                this.startFetch();
+            const c2 = await this.fetchLatestNonOpenedEvents({maxEvents: 10000 - c1});
```

Comment:
Am I understanding this correctly? If we fetch 10k opened events, we won't run `fetchLatestNonOpenedEvents` or `fetchMissing`?

Reply:
Yeah, correct. I realize this part isn't different. The key is I added db persistence to the `fetchMissing` and bumped back the fallback following boot to help account for the rollout.

Previously I had `maxEvents: Infinity` for `fetchLatestOpenedEvents`, which we could go back to, as this just keeps looping on 10k. I had it looping so that it's re-cycling around every 5 mins. That part is kinda semantics.. it's the same behavior just a bit more logging. I'm not bullish on either approach.